### PR TITLE
`MoveType` recognizes castling moves

### PR DESCRIPTION
**Pull request by swcraig**
If a king is moving more than 1 space, it is possibly attempting to
castle.

### CODE REVIEW

**Comment by marclave** [comment]
Typo in the PR name

File: spec/move_type_spec.rb [hunk 1]
```diff
@@ -49,5 +51,29 @@
 
       it { is_expected.to be_a PawnPromotion }
     end
+
+    context 'when it is a castling move' do
+      let(:move) do
+        PlayerRequestedMove.new(board, player_one, cell_e1, cell_b1)
+      end
+
+      before do
```
Comment (jacquesporveau) [inline spec/move_type_spec.rb]: Let's one line this.

File: spec/move_type_spec.rb [hunk 2]
```diff
@@ -49,5 +51,29 @@
...
+
+      it { is_expected.to be_a CastlingMove }
+    end
+
+    context 'when it is a king moving one cell' do
+      let(:move) do
+        PlayerRequestedMove.new(board, player_one, cell_e1, cell_e2)
+      end
+
+      before do
```
Comment (jacquesporveau) [inline spec/move_type_spec.rb]: Here too.

File: lib/move_types/castling_move.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# For when a player wants to perform a castling move
+# https://en.wikipedia.org/wiki/Castling
```
Comment (jacquesporveau) [inline lib/move_types/castling_move.rb]: :+1:

**Comment by swcraig** [comment]
Discussed over Slack, one-lining those two places would make the lines too long (by one character).

**Pull request merged**